tests/drivers/rtc/rtc_api_helpers/src/test_rtc_time_to_tm.c

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2022 Bjarki Arge Andreasen
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/ztest.h>
+#include <zephyr/drivers/rtc.h>
+#include <time.h>
+
+ZTEST(rtc_api_helpers, validate_rtc_time_compat_with_tm)
+{
+	zassert(offsetof(struct rtc_time, tm_sec) == offsetof(struct tm, tm_sec),
+		"Offset of tm_sec in struct rtc_time does not match struct tm");
```

Comment:
I was suggesting you just add a BUILD_ASSERT in rtc.h to verify these offsets, match but I'm fine with this getting checked as a test suite.

Reply:
I though it would look a bit messy to have it in the RTC header file :)